src/NServiceBus.Core/Serialization/ConfigureSerialization.cs

```diff
@@ -0,0 +1,62 @@
+﻿namespace NServiceBus.Serialization
+{
+    using System;
+    using NServiceBus.Features;
+    using NServiceBus.MessageInterfaces.MessageMapper.Reflection;
+    using NServiceBus.Serializers;
+
+    /// <summary>
+    /// Base class for configuring <see cref="SerializationDefinition"/> features.
+    /// </summary>
+    public abstract class ConfigureSerialization : Feature
+    {
+        /// <inheritdoc />
+        protected ConfigureSerialization()
+        {
+            EnableByDefault();
+            Prerequisite(context => this.IsDefaultSerializer(context) || this.IsAdditionalDeserializer(context), 
+                string.Format("{0} not enabled since serialization definition not detected.", GetType()));
+        }
+
+        /// <summary>
+        /// Called when the features is activated.
+        /// </summary>
+        protected internal sealed override void Setup(FeatureConfigurationContext context)
+        {
+            context.Container.ConfigureComponent<MessageDeserializerResolver>(DependencyLifecycle.SingleInstance);
+            context.Container.ConfigureComponent<MessageMapper>(DependencyLifecycle.SingleInstance);
+
+            var serializerType = GetSerializerType(context);
+            if (serializerType == null)
+            {
+                return;
+            }
+
+            RegisterSerializer(context, serializerType);
+
+            if (this.IsDefaultSerializer(context))
+            {
+                context.Container.ConfigureProperty<MessageDeserializerResolver>(resolver => resolver.DefaultSerializerType, serializerType);
```

Comment:
Because `MessageDeserializerResolver` is a singleton. You could just use the func builder overload and pass in all into the ctor like this?

```
context.Container.ConfigureComponent<MessageDeserializerResolver>(b => new MessageDeserializerResolver(b.BuildAll<IMessageSerializer>(), defaultSerializer). DependencyLifecycle.SingleInstance);
```
Haven't tested it. Would that work and remove the necessity to have ctor and prop injection?